### PR TITLE
Encounter AI updates and Hodir

### PR DESCRIPTION
**Pull request by rosenrusinov**
- added support for critical damage multiplier buffs like storm power in hodir encounter
- Made changes to the target ai to support it running in individual sims without tank targets
- Made sure all our encounters work with the changes
- Added a new encounter preset for Hodir
- Implemented most of the hodir encounter abilities and added basic ai for both individual and raid sims (still a work in progress)

### CODE REVIEW

**Comment by dbyena** [comment]
When do you expect on getting this merged?

**Comment by rosenrusinov** [comment]
> When do you expect on getting this merged?

Hopefully soon. I need someone to look into the front end part of the PR as im not a front end developer so might be a way to do it better

File: sim/core/unit.go [hunk 1]
```diff
@@ -160,6 +160,27 @@ func (character *Character) DoNothing() {
 	character.doNothing = true
 }
 
+func (unit *Unit) IsActive() bool {
```
Comment (jimmyt857) [inline sim/core/unit.go:163]: `IsEnabled` has already been moved into `Unit`.

File: sim/core/stats/stats.go [hunk 2]
```diff
@@ -332,6 +332,9 @@ type PseudoStats struct {
 	DamageDealtMultiplier       float64            // All damage
 	SchoolDamageDealtMultiplier [SchoolLen]float64 // For specific spell schools (arcane, fire, shadow, etc).
 
+	// Apply Aura: Mod Melee Critical Damage % (Arcane, Fire, Frost, Holy, Nature, Physical, Shadow)
+	SchoolCritMultiplier [SchoolLen]float64 // For specific spell schools (arcane, fire, shadow, etc).
```
Comment (jimmyt857) [inline sim/core/stats/stats.go:336]: I don't see this used anywhere

Comment (rosenrusinov) [reply sim/core/stats/stats.go:336]: Hodir Stormpower buff the comment is how the wow db has it listed in the aura effects

File: sim/core/unit.go (hunk 1 above)
Comment (rosenrusinov) [reply sim/core/unit.go:163]: Did not see that will use it instead 👍

File: sim/core/stats/stats.go (hunk 2 above)
Comment (rosenrusinov) [reply sim/core/stats/stats.go:336]: Just noticed The hodir_ai file is not shown by default from github due to its size so probably missed it

**Pull request merged**